### ISSUE TITLE
Godot crashes when using the tilemap node picker tool.

### ISSUE THREAD

**Issue by Torguen** [godotengine/godot#85776]
### Tested versions

v3.6.beta3.official [21ab700f2]
_bugsquad edit_ 3.5.2 stable

### System information

w10 64

### Issue description

Simply use the picker tool from tilemap mode indiscriminately ( use the shortcut key I ).
After a few actions the editor will have crashed and will not be able to assign values to the nodes or save the scene, among other things.

Simply use the picker tool ( use the shortcut key I ) many times as you can see in the video.
You will see that after several uses of the picker tool, I can now no longer assign a value to the sprite node.

If you continue doing this, the editor may close suddenly.

https://github.com/godotengine/godot/assets/19652075/3184bf7c-8ffd-40dc-b994-9ab699a14812

### Steps to reproduce

Simply use the picker tool from tilemap mode indiscriminately ( use the shortcut key I ).

### Minimal reproduction project (MRP)

[tilemapCrash.zip](https://github.com/godotengine/godot/files/13578410/tilemapCrash.zip)

**Comment by Torguen** [comment]
I have updated the example project since it is not necessary to have two tilemap nodes.
The problem only involves the pick tile tool of the tilemap node.

**Comment by lawnjelly** [comment]
Just looking at this a little. I didn't think I could reproduce it at first, however sure enough after a while you couldn't change the offset. It seemed to carry on running with no errors reported, but when choosing to quit I got:

```
ERROR: Condition "action_level > 0" is true.
   at: clear_history (./core/undo_redo.cpp:348)
```

And a call stack which may have been a crash in the debugger:
```
1  HashMap<Variant, List<Pair<Variant const *, Variant>, DefaultAllocator>::Element *, VariantHasher, VariantComparator, (unsigned char)3, (unsigned char)8>::getptr hash_map.h            317  0x51fb218 
2  HashMap<Variant, List<Pair<Variant const *, Variant>, DefaultAllocator>::Element *, VariantHasher, VariantComparator, (unsigned char)3, (unsigned char)8>::has    hash_map.h            276  0x51fb31d 
3  OrderedHashMap<Variant, Variant, VariantHasher, VariantComparator, (unsigned char)3, (unsigned char)8>::has                                                       ordered_hash_map.h    244  0x51beae1 
4  Dictionary::has                                                                                                                                                   dictionary.cpp        127  0x50e5664 
5  Object::has_meta                                                                                                                                                  object.cpp            1054 0x5134428 
6  SceneTreeEditor::_add_nodes                                                                                                                                       scene_tree_editor.cpp 196  0x3b608ad 
7  SceneTreeEditor::_update_tree                                                                                                                                     scene_tree_editor.cpp 574  0x3b604bc 
8  MethodBind1<bool>::call                                                                                                                                           method_bind.gen.inc   764  0x2a7dcd6 
9  Object::call                                                                                                                                                      object.cpp            918  0x5133c69 
10 MessageQueue::_call_function                                                                                                                                      message_queue.cpp     267  0x5128cd5 
11 MessageQueue::flush                                                                                                                                               message_queue.cpp     358  0x512999a 
12 SceneTree::iteration                                                                                                                                              scene_tree.cpp        593  0x42d58a3 
13 Main::iteration                                                                                                                                                   main.cpp              2362 0x287c216 
14 OS_X11::run                                                                                                                                                       os_x11.cpp            4047 0x284e819 
15 main                                                                                                                                                              godot_x11.cpp         59   0x283ac7c 
```
Will try sanitizer build.

**Comment by Torguen** [comment]
The editor is still running and everything seems to work but some things have crashed because of that bug under the hood. For example, in the video you can see how it is impossible to change a node value.

**Comment by lawnjelly** [comment]
> The editor is still running and everything seems to work but some things have crashed because of that bug under the hood. For example, in the video you can see how it is impossible to change a node value.

Yes there is a bug, looks like a dangling pointer. The `asan` build helps pinpoint it. I may be able to take a look next week if no one else gets to it first.

These areas of the editor I'm not super familiar with, but should be possible to get to the bottom of it.

**Comment by Torguen** [comment]
Perfect, I'll be attentive if you need any test or anything

**Comment by lawnjelly** [comment]
The asan finding above was a separate bug I think, due to the `edited_scene_root` being out of date, I have a possible fix for that.

The main issue problem appears to be in the `undo_redo` system, specifically `TileMapEditor::forward_gui_input` calls `_start_undo()` which creates an undo action, but that may never end up getting committed (which I think should happen for each action). As a result the queue is getting blown I think.

Will look more tomorrow. :+1:

**Comment by akien-mga** [comment]
Fixed by #88871.
